vcluster/integrations/custom-plugins.mdx

```diff
@@ -0,0 +1,17 @@
+---
+title: Custom Plugins
+sidebar_label: Custom Plugins
+sidebar_class_name: host-nodes
+sidebar_position: 4
+---
+
+:::tip Recommended Reads
+If you want to start developing your own vCluster plugins, read about [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) and [kube builder](https://book.kubebuilder.io/introduction.html),  which uses the controller runtime internally.
+:::
+
+vCluster provides a [SDK](https://github.com/loft-sh/vcluster-sdk) for writing plugin controllers that abstracts a lot of the syncer complexity away from the user, but still gives you access to the underlying structures if you need it.
+Internally, the vCluster SDK uses the [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) project, which is used by vCluster itself to create the controllers.
+The vCluster SDK lets you write custom plugin controllers with just a few lines of code.
+
+Since the plugin SDK interfaces are mostly compatible with the vCluster syncers, you can also take a look at how those are implemented in [the vCluster itself](https://github.com/loft-sh/vcluster/tree/main/pkg/controllers/resources), which work in most cases the same way as if those would be implemented in a plugin.
+It would be even possible to reimplement all vCluster syncers in a separate plugin.
```

Comment:
Maybe adding a link to the configuration page would help users find where to configure their custom plugins